bin/manager_start.sh

```diff
@@ -27,26 +23,44 @@ JAVA_OPTS="-XX:MaxDirectMemorySize=$MAX_DIRECT_MEMORY
 
 MAIN="spark.jobserver.JobManager"
 
-MESOS_OPTS=""
-if [ $1 == "mesos-cluster" ]; then
-    MESOS_OPTS="--master $MESOS_SPARK_DISPATCHER --deploy-mode cluster"
-    appdir=$REMOTE_JOBSERVER_DIR
-fi
+# copy files via spark-submit and read them from current (container) dir
+if [ $2 = "cluster" -a -z "$REMOTE_JOBSERVER_DIR" ]; then
+  SPARK_SUBMIT_OPTIONS="$SPARK_SUBMIT_OPTIONS
+    --master $1 --deploy-mode cluster
+    --files $appdir/log4j-cluster.properties,$conffile"
```

Comment:
Does the --files option handle a directory?  If so we can package all the files inside a subdir of $appdir.  Maybe not worth the change given there are only two files.

Reply:
Not sure how directories are handled. What additional files do we need? The job-server.jar + spark jar and all other one in classpath are already uploaded in a YARN setup (no need to add them again). The spark standanlone mode behaves different and did not upload any files in cluster mode...